data/test/constraints/valid_values/all_types.isl

```diff
@@ -0,0 +1,48 @@
+type::{
+  valid_values: [
+    null,
+    true,
+    2,
+    4e0,
+    5.,
+    2018T,
+    hello,
+    "hello",
+    {{"hello"}},
+    {{aGVsbG8=}},
+    [1, 2, 3],
+    (4 5 6),
+    {a: 7, b: 8, c: 9},
+  ],
+  type: $any,   // overrides default "type: any"
```

Comment:
Could explain the difference between `any` and `$any`? Does it boil down to nullability?

From the spec:
`$any`
> The Ion types are prefixed with $, and correspond precisely with the types defined by the Ion data model, including strongly-typed null values
> ....
> $any: represents any of the Ion types as well as types derived from [ion] types

`any`
> Unless otherwise specified, type definitions have an implicit constraint type: any, and thereby represent any non-null value from the universe of values representable in the Ion data model.